spec/controllers/goals_controller_spec.rb

```diff
@@ -30,4 +30,70 @@
       expect(response).to redirect_to(user_path(my_user.id))
     end
   end
+
+  describe "DELETE #destroy" do
+    it "responds with http status 200 ok" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+
+      sign_in(user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(response).to have_http_status(200)
+    end
+
+    it "deletes the expected goal" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+
+      sign_in(user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(Goal.count).to eq(0)
+    end
+
+    it "deletes only the expected goal" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+      second_goal = create(:goal, user: user)
+
+      sign_in(user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(Goal.count).to eq(1)
+    end
+
+    it "does not delete goal belonging to another user" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+      other_user = create(:user, email: "someone.else@gmail.com")
+
+      sign_in(other_user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(Goal.count).to eq(1)
+    end
+
+    xit "sets the successful flash message" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+
+      sign_in(user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(page).to have_content("Goal completed successfully!")
```

Comment:
`page` and `have_content` are limited to usage in feature specs.

To test the flash here, borrow from the controller spec cheatsheet: https://gist.github.com/eliotsykes/5b71277b0813fbc0df56#file-rspec_cheatsheet_controller_spec-rb-L62-L64